dev/com.ibm.ws.logging.osgi/src/com/ibm/ws/logging/internal/osgi/TrOSGiLogForwarder.java

```diff
@@ -80,14 +89,19 @@ OSGiTraceComponent getTraceComponent(Bundle b) {
                     ffdcMe = bsn + "-" + b.getVersion();
                 }
                 String logName = id + "-" + bsn;
-                String group = b.getHeaders("").get("WS-TraceGroup");
-                String[] groups;
-                if (group == null) {
-                    groups = new String[] { bsn, LOG_SERVICE_GROUP, TRACE_SPEC_OSGI_EVENTS };
-                } else {
-                    groups = new String[] { bsn, group, LOG_SERVICE_GROUP, TRACE_SPEC_OSGI_EVENTS };
+                List<String> groups = new ArrayList<>(5);
+                groups.add(bsn);
+                groups.add(LOG_SERVICE_GROUP);
+                groups.add(TRACE_SPEC_OSGI_EVENTS);
+                String groupHeader = b.getHeaders("").get("WS-TraceGroup");
+                if (groupHeader != null) {
+                    groups.add(groupHeader);
```

Comment:
Any idea why Brent prioritized the group header as the second list item rather than the fourth?  Perhaps due to its the affinity to the bundle, which would be apparent when dumping groups in a trace event.

Reply:
No idea.  I don't think order matters here, but I can move this up to after the BSN if you prefer.

Reply:
No need.  I would have otherwise asked for the change.  Just odd that the order seemed important at some point.

Reply:
> No need. I would have otherwise asked for the change. Just odd that the order seemed important at some point.

I'll change it back to avoid concern.  I don't think order ever really mattered here, but I also have no proof to back up that belief.